### PR TITLE
select: add flags and warnings.

### PR DESCRIPTION
**Pull request by coocood**
To store required information to properly handle truncate error.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @BusyJay PTAL

File: proto/select.proto [hunk 1]
```diff
@@ -66,6 +66,14 @@ message SelectRequest {
...
 	// time zone offset in seconds
 	optional int64 time_zone_offset = 14;
+
+	/* flags is used to store flags that change the execution mode, it contains:
+		truncate_as_error = 1
+			Truncate wrong value will return error if set, return warning if not set.
+		...
+		add more when needed.
+	*/
+	optional uint64 flags = 15;
```
Comment (BusyJay) [inline proto/select.proto]: Why u64?

Comment (shenli) [inline proto/select.proto]: add [(gogoproto.nullable) = false];?

Comment (coocood) [inline proto/select.proto]: The zero value behavior is different than current behavior, so keep it nullable is more compatible.

**Comment by coocood** [comment]
@shenli @BusyJay PTAL

**Comment by shenli** [comment]
LGTM

File: proto/select.proto [hunk 2]
```diff
@@ -65,7 +65,17 @@ message SelectRequest {
 	repeated Expr aggregates = 13;
 
 	// time zone offset in seconds
-	optional int64 time_zone_offset = 14;
+	optional int64 time_zone_offset = 14 [(gogoproto.nullable) = false];
+
+	/* flags is used to store flags that change the execution mode, it contains:
```
Comment (BusyJay) [inline proto/select.proto]: All the comment in this file is prefixed will `//`, I think here should follow the convention too.

File: proto/select.proto [hunk 3]
```diff
@@ -65,7 +65,17 @@ message SelectRequest {
...
+		ignore_truncate = 1
+			truncate error should be ignore if set.
+		truncate_as_warning = 1 << 1
+			when ignored_truncate is not set, return warning instead of error if this flag is set.
+		...
+		add more when needed.
+	*/
+	optional uint64 flags = 15 [(gogoproto.nullable) = false];
```
Comment (BusyJay) [inline proto/select.proto]: Why use plural?

Comment (coocood) [inline proto/select.proto]: search github code "long flags;" has much more results than ""long flag;"

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**